core/commonMain/src/ArgParser.kt

```diff
@@ -639,7 +643,11 @@ open class ArgParser(
             }
             // Parse arguments for subcommand.
             usedSubcommand?.let {
-                it.parse(subcommandsOptions + listOfNotNull("--".takeUnless { treatAsOption }) + subcommandsArguments)
+                if (strictSubcommandOptionsOrder) {
+                    it.parse(args.slice(argIterator.nextIndex() until args.size))
```

Comment:
It seems that flag isn't propagated to the next subcommands, so if there is several levels of subcommands then the behaviour differ. It should be done like for prefix setting https://github.com/Kotlin/kotlinx-cli/blob/29b1e8961e4b5f6b62bfd9c726ae3512715b2674/core/commonMain/src/ArgParser.kt#L333.

Reply:
That's right, agree